diff --git a/metatomic-torch/CMakeLists.txt b/metatomic-torch/CMakeLists.txt
index d54f7634..93a0c0e5 100644
--- a/metatomic-torch/CMakeLists.txt
+++ b/metatomic-torch/CMakeLists.txt
@@ -85,8 +85,6 @@ if (TARGET metatensor_torch)
     check_compatible_versions(${METATENSOR_TORCH_BUILD_VERSION} ${REQUIRED_METATENSOR_TORCH_VERSION})
 else()
     find_package(metatensor_torch ${REQUIRED_METATENSOR_TORCH_VERSION} CONFIG REQUIRED)
-    # No version specification here, we'll use whatever `metatensor_torch` uses
-    find_package(metatensor CONFIG REQUIRED)
 endif()
 
 # FindCUDNN.cmake distributed with PyTorch is a bit broken, so we have a
@@ -118,7 +116,7 @@ set_target_properties(metatomic_torch PROPERTIES
     BUILD_VERSION ${METATOMIC_TORCH_FULL_VERSION}
 )
 
-target_link_libraries(metatomic_torch PUBLIC torch metatensor_torch metatensor ${CMAKE_DL_LIBS})
+target_link_libraries(metatomic_torch PUBLIC torch metatensor_torch ${CMAKE_DL_LIBS})
 target_compile_features(metatomic_torch PUBLIC cxx_std_17)
 target_include_directories(metatomic_torch PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -195,15 +193,7 @@ FetchContent_Declare(nlohmann_json
     URL https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz
     URL_HASH SHA256=d6c65aca6b1ed68e7a182f4757257b107ae403032760ed6ef121c9d55e81757d
 )
-if (CMAKE_VERSION VERSION_GREATER 3.18)
-    FetchContent_MakeAvailable(nlohmann_json)
-else()
-    if (NOT nlohmann_json_POPULATED)
-        FetchContent_Populate(nlohmann_json)
-    endif()
-
-    add_subdirectory(${nlohmann_json_SOURCE_DIR} ${nlohmann_json_BINARY_DIR})
-endif()
+FetchContent_MakeAvailable(nlohmann_json)
 
 target_link_libraries(metatomic_torch PRIVATE nlohmann_json::nlohmann_json)
 
